client/lib/format-number-compact/index.js

```diff
@@ -9,10 +13,15 @@ import { THOUSANDS } from './thousands';
  */
 export default function formatNumberCompact( number, code = i18n.getLocaleSlug() ) {
 	//use numberFormat directly from i18n in this case!
-	if ( isNaN( number ) || ! THOUSANDS[ code ] ) {
+	if ( isNaN( number ) ) {
 		return null;
 	}
 
+	// Rely on `formatNumberMetric` we don't support the language, or when the number is 1M+
+	if ( ! THOUSANDS[ code ] || number >= ONE_M ) {
+		return formatNumberMetric( number );
+	}
```

Comment:
FYI @Automattic/i18n about this change to fix `formatNumberCompact` with numbers which are 1M+. Someone created the simpler `formatNumberMetric` which on the other hand doesn't take into account localization. These should be combined.

Note that someone also _copied_ the files to [component in the package](https://github.com/Automattic/wp-calypso/blob/6d0cd83763d613e659aa36719cec97cffa32e205/packages/components/src/count/format-number-compact.js).

So overall these files look like they could do with a cleanup.

Reply:
To be honest, I wasn't aware of the existence of this function and I haven't looked into the details, but at first glance it seems like it can be replaced by Intl.NumberFormat with the compact notation, i.e. `Intl.NumberFormat( locale, { notation: 'compact' } ).format( number );`.

Reply:
Do we have something to handle WP.com locale -> locales expected by the `Intl` somewhere?

Reply:
Not that I'm aware of, but I know that we're already using the `Intl` with Calypso's language codes (most of which are aligned with ISO-639-1) in various places and it works correctly.

Reply:
I would remove `formatNumberCompact` and `formatNumberMetric` functions completely and replace their usage with `Intl.NumberFormat` as @yuliyan suggested. WDYT @simison?

Those functions are not widely used, only in a couple of places.

Reply:
Works for me 👍

Reply:
Just found a helper function that we can use:
https://github.com/Automattic/wp-calypso/blob/trunk/packages/components/src/number-formatters/lib/format-number.ts

I will use it instead of call `Intl.NumberFormat` directly.

Reply:
@pkuliga, I noticed that the helper function defaults to using the browser locale, which may be different than the user locale, so just make sure to pass the latter.

Reply:
Sure, thanks!